maven/maven.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: maven
+spec:
+  inputs:
+    params:
+    - name: GOALS
+      description: The Maven goals to run
+      type: array
+      default:
+      - "package"
+    - name: MAVEN_MIRROR_URL
+      description: The Maven bucketrepo- mirror
+      type: string
+      default: ""
+    resources:
+    - name: source
+      targetPath: /
+      type: git
+  steps:
+    - name: mvn-settings
+      image: registry.redhat.io/ubi8/ubi-minimal
+      workingdir: /.m2
+      command:
+        - '/bin/bash'
+        - '-c'
+      args:
+        - |-
+          [[ -n '$(inputs.params.MAVEN_MIRROR_URL)' ]] && \
+          cat > /.m2/settings.xml <<EOF
```

Comment:
@nikhil-thomas could the user provide their own `.settings.xml` by mounting a configmap at this location? If so then we should also check if there exists a `.settings.xml` and not overwrite it.

Reply:
@siamaksade 
is this what we need 

https://github.com/nikhil-thomas/catalog/blob/d562bea2842a92f11036d364082aa87ccd585dad/maven/maven.yaml#L30

and 

https://github.com/nikhil-thomas/catalog/tree/add-maven-task/maven#with-custom-m2settingsyaml

Reply:
Yes, exactly 👍